### PR TITLE
Update healthcheck info live

### PR DESCRIPTION
**Pull request by davepagurek**
- moves all websocket authentication into its own controller
- Makes a second websocket channel for each worker. `worker.${id}` is to send spawn requests to the worker itself, `worker_info.${id}` is to subscribe to health check updates on the ui.

### CODE REVIEW

File: app/controllers/websocket_authentication_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
```
Comment (AndrewMcBurney) [inline app/controllers/websocket_authentication_controller.rb]: I like having a separate controller a lot better than having a public method in the `job_connection_controller`. Nice work on separation of concerns! 💯

File: app/controllers/websocket_authentication_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    job = Job.find_by(id: match[:id].to_i)
+    if job && current_user && job.user == current_user
+      accept_channel job.as_json
+    else
+      deny_channel
+    end
+  end
+
+  def authorize_worker_info
```
Comment (AndrewMcBurney) [inline app/controllers/websocket_authentication_controller.rb]: Hmm, these two methods `authorize_worker_info`, and `authorize_job` are very similar. Do you think we'd be able to create a helper method to reuse? I'm just thinking about if we want to extend this controller for further channels.

Comment (davepagurek) [reply app/controllers/websocket_authentication_controller.rb]: Refactored, now it passes the channel prefix and the class to search for into a method

File: app/controllers/websocket_authentication_controller.rb [hunk 3]
```diff
@@ -14,26 +14,20 @@ def authorize_channels
...
+    authorize_by_id("worker_info", Worker)
+  end
+
+  def authorize_by_id(channel_prefix, klass)
+    match = /^#{channel_prefix}\.(?<id>\w+)$/.match(message[:channel])
     return deny_channel unless match
 
-    worker = Worker.find_by(id: match[:id].to_i)
-    if worker && current_user && worker.user == current_user
-      accept_channel
```
Comment (AndrewMcBurney) [inline app/controllers/websocket_authentication_controller.rb]: Just one comment on the diff. 

Before you had `accept_channel` for the worker, and not `accept_channel worker.as_json`. Will this new logic still have the same desired functionality for the `authorize_by_id("worker_info", Worker)` with the `instance.as_json` being passed into the `accept_channel` method?

File: app/models/worker.rb [hunk 4]
```diff
@@ -12,7 +13,23 @@ def update_heartbeat
...
+  end
+
+  def health_info
+    as_json.slice("id", *healthcheck_params)
+  end
+
   def channel
     WebsocketRails["worker.#{id}"]
   end
+
```
Comment (AndrewMcBurney) [inline app/models/worker.rb]: Can we make the methods below this line private?

Comment (AndrewMcBurney) [inline app/models/worker.rb]: Can we make the methods below this line private?

File: app/controllers/worker_connection_controller.rb [hunk 5]
```diff
@@ -8,7 +8,8 @@ def connect
 
   def healthcheck
     Rails.logger.info "Set healthcheck for worker: #{message['id']}."
-    worker.update(message.slice(*worker_healthcheck_params))
+    worker&.update(message.slice(*worker.healthcheck_params))
```
Comment (AndrewMcBurney) [inline app/controllers/worker_connection_controller.rb]: Nice refactor, putting the healthcheck_params in the `Worker` model

File: config/events.rb [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 WebsocketRails::EventMap.describe do
   namespace :websocket_rails do
-    subscribe :subscribe_private, to: JobConnectionController, with_method: :authorize_channels
+    subscribe :subscribe_private, to: WebsocketAuthenticationController, with_method: :authorize_channels
```
Comment (AndrewMcBurney) [inline config/events.rb]: I like this a lot better, good call. 👍

File: app/controllers/websocket_authentication_controller.rb (hunk 3 above)
Comment (davepagurek) [reply app/controllers/websocket_authentication_controller.rb]: yeah, just sends extra info with the initial state, which we can use or ignore as we feel like on the frontend.

**Pull request merged**